veracruz-mcu-client/main.c

```diff
@@ -0,0 +1,153 @@
+/*
+ * A Veracruz client targetting microcontroller devices
```

Comment:
I think this is related to @dreemkiller's questions about about the relationship between the demo and the MCU client itself: whilst this header indicates that this file implements a client, it seems to be entirely demo related.  Can we cleanly separate the two, including moving the demo out of the directory where the client code is defined, or similar?